### ISSUE TITLE
Blender throws a KeyError when trying to export to GLB

### ISSUE THREAD

**Issue by richard-gebbia** [KhronosGroup/glTF-Blender-IO#1877]
**Describe the bug**
I can't seem to export a model to GLB format. When doing so I get a Python stack trace that leads to a KeyError.

**To Reproduce**
1. Open supplied .blend file
2. Go to File > Export > glTF 2.0 (.glb/.gltf)
3. Set "Format" to "glTF Binary (.glb)"
4. Check "Transform > "+Y Up"
5. Click blue "Export glTF 2.0" button at bottom
6. See error message

**Expected behavior**
No error message should appear and the file is exported successfully to GLB format.

**Screenshots**
Trace of error message:
```
Python: Traceback (most recent call last):
  File "C:\Program Files\Blender Foundation\Blender 3.4\3.4\scripts\addons\io_scene_gltf2\__init__.py", line 693, in execute
    return gltf2_blender_export.save(context, export_settings)
  File "C:\Program Files\Blender Foundation\Blender 3.4\3.4\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_export.py", line 35, in save
    json, buffer = __export(export_settings)
  File "C:\Program Files\Blender Foundation\Blender 3.4\3.4\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_export.py", line 52, in __export
    __gather_gltf(exporter, export_settings)
  File "C:\Program Files\Blender Foundation\Blender 3.4\3.4\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_export.py", line 66, in __gather_gltf
    active_scene_idx, scenes, animations = gltf2_blender_gather.gather_gltf2(export_settings)
  File "C:\Program Files\Blender Foundation\Blender 3.4\3.4\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather.py", line 34, in gather_gltf2
    animations += __gather_animations(blender_scene, export_settings)
  File "C:\Program Files\Blender Foundation\Blender 3.4\3.4\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather.py", line 93, in __gather_animations
    animations_, merged_tracks = gltf2_blender_gather_animations.gather_animations(obj_uuid, merged_tracks, len(animations), export_settings)
  File "C:\Program Files\Blender Foundation\Blender 3.4\3.4\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_animations.py", line 114, in gather_animations
    animation = __gather_animation(obj_uuid, blender_action, export_settings)
  File "C:\Program Files\Blender Foundation\Blender 3.4\3.4\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_animations.py", line 164, in __gather_animation
    channels=__gather_channels(obj_uuid, blender_action, export_settings),
  File "C:\Program Files\Blender Foundation\Blender 3.4\3.4\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_animations.py", line 201, in __gather_channels
    return gltf2_blender_gather_animation_channels.gather_animation_channels(
  File "C:\Program Files\Blender Foundation\Blender 3.4\3.4\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_cache.py", line 38, in wrapper_cached
    result = func(*args, **kwargs)
  File "C:\Program Files\Blender Foundation\Blender 3.4\3.4\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_animation_channels.py", line 130, in gather_animation_channels
    channel = gather_animation_channel(obj_uuid, channel_group, export_settings, None, None, bake_range_start, bake_range_end, force_range, blender_action.name, None, True)
  File "C:\Program Files\Blender Foundation\Blender 3.4\3.4\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_animation_channels.py", line 300, in gather_animation_channel
    sampler = __gather_sampler(channels, obj_uuid, export_settings, bake_bone, bake_channel, bake_range_start, bake_range_end, force_range, action_name, driver_obj_uuid, node_channel_is_animated)
  File "C:\Program Files\Blender Foundation\Blender 3.4\3.4\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_animation_channels.py", line 368, in __gather_sampler
    return gltf2_blender_gather_animation_samplers.gather_animation_sampler(
  File "C:\Program Files\Blender Foundation\Blender 3.4\3.4\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_cache.py", line 38, in wrapper_cached
    result = func(*args, **kwargs)
  File "C:\Program Files\Blender Foundation\Blender 3.4\3.4\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_animation_samplers.py", line 59, in gather_animation_sampler
    input = __gather_input(channels, obj_uuid, is_armature, non_keyed_values,
  File "C:\Program Files\Blender Foundation\Blender 3.4\3.4\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_cache.py", line 38, in wrapper_cached
    result = func(*args, **kwargs)
  File "C:\Program Files\Blender Foundation\Blender 3.4\3.4\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_animation_samplers.py", line 243, in __gather_input
    keyframes, is_baked = gltf2_blender_gather_animation_sampler_keyframes.gather_keyframes(blender_obj_uuid,
  File "C:\Program Files\Blender Foundation\Blender 3.4\3.4\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_cache.py", line 38, in wrapper_cached
    result = func(*args, **kwargs)
  File "C:\Program Files\Blender Foundation\Blender 3.4\3.4\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_animation_sampler_keyframes.py", line 357, in gather_keyframes
    mat = get_object_matrix(blender_obj_uuid,
  File "C:\Program Files\Blender Foundation\Blender 3.4\3.4\scripts\addons\io_scene_gltf2\blender\exp\gltf2_blender_gather_cache.py", line 93, in wrapper_objectcache
    return result[cache_key_args[0]][cache_key_args[1]][cache_key_args[4]]
KeyError: 1.0
```

**.blend file/ .gltf**
[Archer.zip](https://github.com/KhronosGroup/glTF-Blender-IO/files/10939511/Archer.zip)

**Version**
 - OS: Windows 10 Pro 21H2
 - Blender Version: 3.4.1

**Comment by julienduroure** [comment]
Hello,
Thanks for the report.
This is a duplicate of #1803 

Linked to frame that are not integer.
Workaround for now : select all your keyframes in dopsheet, and Key > Snap > Snap Selection to nearest frame

This is already fixed in main branch (that will be Blender 3.6)

**Comment by richard-gebbia** [comment]
Thank you very much. This fixed the issue for me, and I'll be sure to get Blender 3.6 when it comes out.
